ledger-tool/src/main.rs

```diff
@@ -1652,22 +1717,21 @@ fn main() {
                             .ok();
                     }
 
-                    if let Some(tx_receiver) = tx_receiver {
-                        tx_receiver.join().unwrap();
-                    }
-
-                    if let Some(recorded_slots_file) = record_slots_file {
-                        if let Ok(recorded_slots) = recorded_slots.clone().unwrap().lock() {
-                            let bank_hashes =
-                                bank_hash_details::BankHashDetails::new(recorded_slots.to_vec());
-
-                            // writing the json file ends up with a syscall for each number, comma, indentation etc.
-                            // use BufWriter to speed things up
+                    if let Some(slot_recorder_config) = slot_recorder_config {
+                        if let Some(transaction_recorder) =
+                            slot_recorder_config.transaction_recorder
+                        {
+                            transaction_recorder.join().unwrap();
+                        }
 
-                            let writer = std::io::BufWriter::new(recorded_slots_file);
+                        let slot_details = slot_recorder_config.slot_details.lock().unwrap();
+                        let bank_hashes =
+                            bank_hash_details::BankHashDetails::new(slot_details.to_vec());
```

Comment:
I'll test this separately, but I believe that we should be able to do a series of `into_inner()`'s to extract the vector out of `Arc` and `Mutex`; this would avoid the `to_vec()`